[Turbopack]: fails to resolve imports with node subpath aliases

### Link to the code that reproduces this issue

https://github.com/tarungunampalli-rox/turbopack-imports-repro

### To Reproduce

1. Add `imports` field to package.json 
```
  "imports": {
    "#/*": "./src/*"
  },
```
2. Import some internal src file starting with the `#` subpath alias
3. run build with turbopack

### Current vs. Expected behavior

Logs from the reproduction show
```
Error: Turbopack build failed with 2 errors:
./src/app
Module not found: Can't resolve '#/greeting'
https://nextjs.org/docs/messages/module-not-found
./src/app/page.tsx:1:1
Module not found: Can't resolve '#/greeting'
```

I expect turbopack should be able to resolve internal imports using the `#/` subpath alias


### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 25.5.0: Mon Apr 27 20:41:12 PDT 2026; root:xnu-12377.121.6~2/RELEASE_ARM64_T6050
  Available memory (MB): 36864
  Available CPU cores: 18
Binaries:
  Node: 24.15.0
  npm: 11.12.1
  Yarn: 1.22.22
  pnpm: 11.1.1
Relevant Packages:
  next: 16.3.0-canary.36 // Latest available version is detected (16.3.0-canary.36).
  eslint-config-next: N/A
  react: 19.2.0
  react-dom: 19.2.0
  typescript: 5.9.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Turbopack

### Which stage(s) are affected? (Select all that apply)

next build (local), Vercel (Deployed)

### Additional context

Both latest canary (16.3.0-canary.36) and 16.2.6 are seeing this issue

This [related issue](https://github.com/vercel/next.js/issues/93295) seems like it was only resolved for external imports. Internal imports seem to still not resolve correctly

--- a/turbopack/crates/turbopack-core/src/resolve/mod.rs
+++ b/turbopack/crates/turbopack-core/src/resolve/mod.rs
@@ -3309,11 +3309,6 @@ async fn resolve_package_internal_with_imports_field(
     let Pattern::Constant(specifier) = pattern else {
         bail!("PackageInternal requests can only be Constant strings");
     };
-    // Node.js spec rejects specifiers that are exactly "#", start with "#/", or end with "/".
-    // However, the "#/*" pattern (e.g. { "#/*": "./src/*" }) is widely used in TypeScript
-    // projects and is supported by webpack (enhanced-resolve). Turbopack follows webpack
-    // behavior here to avoid a webpack/Turbopack divergence for this common pattern.
-    // See: https://github.com/vercel/next.js/issues/94290
     if specifier == "#" || specifier.ends_with('/') {
         ResolvingIssue {
             severity: resolve_error_severity(resolve_options).await?,

--- a/turbopack/crates/turbopack-core/src/resolve/mod.rs
+++ b/turbopack/crates/turbopack-core/src/resolve/mod.rs
@@ -3309,8 +3309,12 @@ async fn resolve_package_internal_with_imports_field(
     let Pattern::Constant(specifier) = pattern else {
         bail!("PackageInternal requests can only be Constant strings");
     };
-    // https://github.com/nodejs/node/blob/1b177932/lib/internal/modules/esm/resolve.js#L615-L619
-    if specifier == "#" || specifier.starts_with("#/") || specifier.ends_with('/') {
+    // Node.js spec rejects specifiers that are exactly "#", start with "#/", or end with "/".
+    // However, the "#/*" pattern (e.g. { "#/*": "./src/*" }) is widely used in TypeScript
+    // projects and is supported by webpack (enhanced-resolve). Turbopack follows webpack
+    // behavior here to avoid a webpack/Turbopack divergence for this common pattern.
+    // See: https://github.com/vercel/next.js/issues/94290
+    if specifier == "#" || specifier.ends_with('/') {
         ResolvingIssue {
             severity: resolve_error_severity(resolve_options).await?,
             file_path: file_path.clone(),

--- a/turbopack/crates/turbopack-tests/tests/execution/turbopack/resolving/subpath-imports-slash/input/index.js
+++ b/turbopack/crates/turbopack-tests/tests/execution/turbopack/resolving/subpath-imports-slash/input/index.js
@@ -1,0 +1,5 @@
+import greeting from '#/greeting.js'
+
+it('should resolve #/* subpath imports', () => {
+  expect(greeting).toBe('hello')
+})

--- a/turbopack/crates/turbopack-tests/tests/execution/turbopack/resolving/subpath-imports-slash/input/package.json
+++ b/turbopack/crates/turbopack-tests/tests/execution/turbopack/resolving/subpath-imports-slash/input/package.json
@@ -1,0 +1,6 @@
+{
+  "name": "subpath-imports-slash",
+  "imports": {
+    "#/*": "./src/*"
+  }
+}

--- a/turbopack/crates/turbopack-tests/tests/execution/turbopack/resolving/subpath-imports-slash/input/src/greeting.js
+++ b/turbopack/crates/turbopack-tests/tests/execution/turbopack/resolving/subpath-imports-slash/input/src/greeting.js
@@ -1,0 +1,1 @@
+export default 'hello'

